### PR TITLE
travis: update build to use python3

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: #8383 

_Goal_: the purpose of the pull-request is to verify the build using `python3` instead of EOL `python2`.

The pull-request includes the following changes:
- [x] **ubuntu**: updates the `dist` to [`bionic`](https://docs.travis-ci.com/user/reference/bionic/) which includes [python3.6](https://docs.travis-ci.com/user/reference/bionic/#python-support)
- [x] **macOS**: updates the build to use `python3` by default
- [x] **windows**: installs `python3` through [chocolatey](https://chocolatey.org/packages/python3/3.8.5.20200721)
- [x] cleanup existing builds, removes python2 specific-builds

#### How to test

- the continuous-integration should pass successfully, they should each use a version of `python3`

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I need to investigate why it is currently failing for ` yarn test:electron`:

```
$ travis_retry yarn test:electron
yarn run v1.22.4
$ yarn rebuild:electron && run test "@theia/example-electron"
$ theia rebuild:electron
Processing @theia/node-pty
Processing nsfw
Processing native-keymap
Processing find-git-repositories
- Searching dependency tree
✔ Rebuild Complete
lerna info version 2.11.0
lerna info scope @theia/example-electron
@theia/example-electron: $ electron-mocha --timeout 60000 "./lib/test/**/*.espec.js"
@theia/example-electron: Segmentation fault (core dumped)
@theia/example-electron: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@theia/example-electron: error Command failed with exit code 139.
lerna ERR! execute callback with error
lerna ERR! Error: Command failed: yarn run test
lerna ERR! Segmentation fault (core dumped)
lerna ERR! error Command failed with exit code 139.
lerna ERR! 
lerna ERR! $ electron-mocha --timeout 60000 "./lib/test/**/*.espec.js"
lerna ERR! info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! 
lerna ERR!     at /home/travis/build/eclipse-theia/theia/node_modules/execa/index.js:236:11
Error: Command failed: yarn run test
Segmentation fault (core dumped)
error Command failed with exit code 139.
```

**Comment by marcdumais-work** [comment]
@vince-fugnitto can you uplift this PR so we can benefit from the fix you merged yesterday (https://github.com/eclipse-theia/theia/pull/8454)? I am not sure but it may be related to the failure here.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto can you uplift this PR so we can benefit from the fix you merged yesterday (#8454)? I am not sure but it may be related to the failure here.

Sure!

**Comment by marcdumais-work** [comment]
I think it helped - the Linux build is about to pass.

**Comment by marcdumais-work** [comment]
Nah, still failing in the end...

**Comment by marcdumais-work** [comment]
It's passing on my machine locally but not in Travis CI. Testing on Gitpod, I got a core dump running `yarn test` in the `examples/electron` folder, with a little bit extra info vs the Travis log. The issue seems related to Electron sandboxing: 

```
gitpod /workspace/theia/examples/electron $ yarn test
yarn run v1.22.4
$ electron-mocha --timeout 60000 "./lib/test/**/*.espec.js"
[4069:0902/162553.574483:FATAL:setuid_sandbox_host.cc(158)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /workspace/theia/node_modules/electron/dist/chrome-sandbox is owned by root and has mode 4755.
Trace/breakpoint trap (core dumped)
error Command failed with exit code 133.
```

Maybe we need to pass-along the `--no-sandbox` Electron CLI argument when running the tests?
```
"test": "electron-mocha --timeout 60000 \"./lib/test/**/*.espec.js\""
```

**Comment by marcdumais-work** [comment]
I wonder why no apparent issue on master?

**Comment by vince-fugnitto** [comment]
> It's passing on my machine locally but not in Travis CI. Testing on Gitpod, I got a core dump running `yarn test` in the `examples/electron` folder, with a little bit extra info vs the Travis log. The issue seems related to Electron sandboxing:
> 
> ```
> gitpod /workspace/theia/examples/electron $ yarn test
> yarn run v1.22.4
> $ electron-mocha --timeout 60000 "./lib/test/**/*.espec.js"
> [4069:0902/162553.574483:FATAL:setuid_sandbox_host.cc(158)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /workspace/theia/node_modules/electron/dist/chrome-sandbox is owned by root and has mode 4755.
> Trace/breakpoint trap (core dumped)
> error Command failed with exit code 133.
> ```
> 
> Maybe we need to pass-along the `--no-sandbox` Electron CLI argument when running the tests?
> 
> ```
> "test": "electron-mocha --timeout 60000 \"./lib/test/**/*.espec.js\""
> ```

Yes, I noticed that as well (that it passes locally). Its a bit odd since the electron tests are actually skipped and it caused a segfault. Good idea, I'll update the test to pass along the `--no-sandbox` argument.

**Comment by marcdumais-work** [comment]
One difference between my local laptop and Travis is that I still have python2 on my machine. Maybe it's still somehow needed?

File: examples/electron/package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "watch": "concurrently -n compile,bundle \"theiaext watch --preserveWatchOutput\" \"theia build --watch --mode development\"",
     "start": "theia start --plugins=local-dir:../../plugins",
     "start:debug": "yarn start --log-level=debug",
-    "test": "electron-mocha --timeout 60000 \"./lib/test/**/*.espec.js\""
+    "test": "electron-mocha --no-sandbox --timeout 60000 \"./lib/test/**/*.espec.js\""
```
Comment (marcdumais-work) [inline examples/electron/package.json:63]: Looking quickly I do not think this is how it's done for electron-mocha. Rather it looks like you need to provide a `.js` file that defined the required argument(s).

https://github.com/jprichardson/electron-mocha#webgl-tests

Comment (vince-fugnitto) [reply examples/electron/package.json:63]: I tested before in gitpod (it failed without the flag), and with it it passed successfully: https://github.com/eclipse-theia/theia/pull/8415#issuecomment-685853207. I'll see if it passes the build, and if not I'll take a look at what you proposed.

**Comment by vince-fugnitto** [comment]
Closing since the pull-request is outdated after the migration to `github actions`.
We should still be able to include `python 3` over `python 2` in the new CI/CD.

**Pull request closed without merging**